Pcap++/header/PcapLiveDevice.h

```diff
@@ -234,6 +232,11 @@ namespace pcpp
 			}
 		};
 
+		/**
+		 * Create new PcapLiveDevice. Typically handled by PcapLiveDeviceList but may be manually created for special use cases such
+		 * as capturing from the same interface more than once on platforms that support it
+		 */
+		PcapLiveDevice(pcap_if_t* pInterface, bool calculateMTU, bool calculateMacAddress, bool calculateDefaultGateway);
```

Comment:
@weyrick the reason this c'tor is private is because it requires `pcap_if_t` which is an internal libpcap structure. I looked in pktvisor and it seems it uses [libpcap's `pcap_findalldevs` API](https://github.com/ns1labs/pktvisor/blob/d421cd630b7fc377fbea6a9097cfc8044983fe9d/src/inputs/pcap/PcapInputStream.cpp#L164) to create `PcapLiveDevice` instances. I'm curious what is the use case to use libpcap directly and not use `PcapLiveDeviceList` 🤔

Reply:
@seladb yes the reason is just as it states in the comment there, and as we discussed in the group message I linked: there is a use case to capture multiple times on the same interface, especially when a different bpf filter is specified, so that the kernel can filter traffic efficiently. PcapPlusPlus makes the assumption that a device will only ever be opened and captured once, and this is an architectural problem for this use case.

Reply:
@weyrick did you conclude that libpcap supports multiple concurrent captures? In the group message you mentioned that `pcap_t` might not be thread safe 🤔

Reply:
@seladb Right, the documentation isn't super clear but it's my belief that while `pcap_t` itself shouldn't be shared across threads, you can create `pcap_t` on the same interface in different threads just fine. This is the way I'm using it in pktvisor without a problem currently.